### PR TITLE
Added unit tests to LogMetricsAsync NaN check

### PR DESCRIPTION
**Pull request by aslotte**
## Resolves
NaN

## Description
Adds unit tests to NaN check for `LogMetricsAsync`

### CODE REVIEW

**Comment by aslotte** [comment]
@Brett-Parker feel free to review this PR :)

**Comment by Brett-Parker** [comment]
@aslotte I was actually closer than I thought. Didn't think it was as easy as this;

`var metric = new { F1Score = double.NaN };` 

The act and assert I did the same. Just arrange I kept going round in circles. In future I will just have confidence and submit for comments :)

File: test/MLOps.NET.Tests/EvaluationCycleCatalogTests.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+using MLOps.NET.Utilities;
+using Moq;
+using System;
+using System.Threading.Tasks;
+
+namespace MLOps.NET.Tests
+{
+    [TestCategory("UnitTests")]
+    [TestClass]
+    public class EvaluationCycleCatalogTests
```
Comment (Brett-Parker) [inline test/MLOps.NET.Tests/EvaluationCycleCatalogTests.cs:14]: I think this has been discussed before but cannnot remember what the decision was. Should this be EvaluationCatalogTests? LifeCycle is a different class.

Comment (aslotte) [reply test/MLOps.NET.Tests/EvaluationCycleCatalogTests.cs]: Great catch, that was a typo from copy-pasting the old test class :)

**Comment by aslotte** [comment]
Since we're using reflection to fetch the metric values, a dynamically created instance is sufficient, thus we can just use `var metric = new { F1Score = double.NaN };` 

You're always welcome to open PR's for discussion!

**Pull request merged**